sigal/plugins/nonmedia_files.py

```diff
@@ -0,0 +1,184 @@
+"""Plugin to index non-media files.
+
+This plugin will copy the files into the build tree and generate generic
+thumbnails for the files.  In-browser previews will likely fail, and
+it is up to the theme to provide correct support for downloads.
+
+Settings available as dictionary in ``nonmedia_files_options``:
+
+- ``ext_as_thumb``: Enable simple thumbnail showing ext.
+  Default to ``True``
+- ``ignore_ext``: List of file extensions to ignore.
+  Default to ``[".md"]``
+- ``thumb_bg_color``: Background color for thumbnail. Accepts (r, g, b) tuple.
+  Default to white ``(255, 255, 255)``.
+- ``thumb_font``: Name or path to font file.
+  Default to ``None`` to automatically choose the font.
+- ``thumb_font_color``: Font color for thumbnail. Accepts (r, g, b) tuple.
+  Default to black ``(0, 0, 0)``.
+- ``thumb_font_size``: Font size for thumbnail. Must select a font to apply.
+  Default to ``40``.
+
+"""
+
+import logging
+import os
+
+from PIL import Image as PILImage
+from PIL import ImageDraw, ImageFont
+
+from pilkit.utils import save_image
+
+from sigal import signals
+from sigal import utils
+from sigal.gallery import Media
+from sigal.settings import Status
+
+logger = logging.getLogger(__name__)
+
+
+DEFAULT_CONFIG = {
+    'ext_as_thumb': True,
+    'ignore_ext': ['.md'],
+    'thumb_bg_color': (255, 255, 255),
+    'thumb_font': None,
+    'thumb_font_color': (0, 0, 0),
+    'thumb_font_size': 40,
+}
+
+
+COMMON_MIME_TYPES = {
+    '.azw': 'application/vnd.amazon.ebook',
+    '.csv': 'text/csv',
+    '.epub': 'application/epub+zip',
+    '.pdf': 'application/pdf',
+    '.svg': 'image/svg+xml',
+    '.txt': 'text/plain',
+    '.zip': 'application/zip',
+}
+
+
+def get_mime(ext):
+    if ext in COMMON_MIME_TYPES:
+        return COMMON_MIME_TYPES[ext]
+    return 'application/octet-stream'
+
+
+class NonMedia(Media):
+    """Gather all informations on a non-media file."""
+
+    type = 'nonmedia'
+
+    def __init__(self, filename, path, settings):
+        super().__init__(filename, path, settings)
+        self.thumb_name = os.path.splitext(self.thumb_name)[0] + '.jpg'
+        self.date = self._get_file_date()
+        self.mime = get_mime(self.src_ext)
+        logger.debug('mime type %s', self.mime)
```

Comment:
`self.mime` doesn't seem to be used ? Or do you think it's useful to have it for the templates ?

Reply:
I added it for the templates.  Not sure if it will be used, but it was simple to add.